lib/ruby_indexer/lib/ruby_indexer/index.rb

```diff
@@ -58,31 +58,35 @@ def register_included_hook(module_name, &hook)
       (@included_hooks[module_name] ||= []) << hook
     end
 
-    sig { params(indexable: IndexablePath).void }
-    def delete(indexable)
-      # For each constant discovered in `path`, delete the associated entry from the index. If there are no entries
-      # left, delete the constant from the index.
-      @files_to_entries[indexable.full_path]&.each do |entry|
-        name = entry.name
-        entries = @entries[name]
-        next unless entries
-
-        # Delete the specific entry from the list for this name
-        entries.delete(entry)
-
-        # If all entries were deleted, then remove the name from the hash and from the prefix tree. Otherwise, update
-        # the prefix tree with the current entries
-        if entries.empty?
-          @entries.delete(name)
-          @entries_tree.delete(name)
-        else
-          @entries_tree.insert(name, entries)
+    sig { params(uri: Uri).void }
+    def delete(uri)
+      path = uri.to_standardized_path
+
+      if path
```

Comment:
When would path be `nil`?

Reply:
For URIs of the untitled scheme (unsaved files), the file name is not a part of the path, it's a part of the URI's opaque.

```ruby
uri = URI("untitled:Untitled-1")
puts uri.path # => nil
puts uri.opaque # => Untitled-1
```